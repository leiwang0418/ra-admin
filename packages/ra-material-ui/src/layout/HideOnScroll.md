HideOnScroll:

```jsx
<HideOnScroll>
    <div>test</div>
</HideOnScroll>
```