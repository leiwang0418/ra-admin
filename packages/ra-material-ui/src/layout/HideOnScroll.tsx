import { Slide, useScrollTrigger } from "@material-ui/core";
import React from "react";
import PropTypes from "prop-types";


const HideOnScroll = (props: HideOnScrollProps) => {
	const { children } = props;
	const trigger = useScrollTrigger();
	return (
		<Slide appear={false} direction="down" in={!trigger}>
			{children}
		</Slide>
	);
};

HideOnScroll.propTypes = {
	children: PropTypes.node.isRequired,
};

export interface HideOnScrollProps {
	children: React.ReactElement;
}

export default HideOnScroll;