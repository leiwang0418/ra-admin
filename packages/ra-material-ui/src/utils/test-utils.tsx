import React, { FC } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { IntlProvider } from "react-intl";
import en from "../compiled-lang/en.json";
import { ThemeProvider } from "@material-ui/styles";
import { createTheme } from "@material-ui/core/styles";

const theme = createTheme();

const customRender = (
	ui: React.ReactElement,
	{ locale = "en", ...renderOptions } = {}
) => {
	const Wrapper: FC = ({ children }) => (
		<IntlProvider locale={locale} messages={en}>
			<ThemeProvider theme={theme}>{children}</ThemeProvider>
		</IntlProvider>
	);

	return render(ui, { wrapper: Wrapper, ...renderOptions });
};

export * from "@testing-library/react";

export { customRender as render };
