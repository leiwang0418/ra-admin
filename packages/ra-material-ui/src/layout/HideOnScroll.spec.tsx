import { render, screen } from "@utils/test-utils";
import HideOnScroll from "./HideOnScroll";

jest.mock("@material-ui/core", () => ({
	...jest.requireActual("@material-ui/core"),
	useScrollTrigger: jest.fn().mockReturnValue(true),
}));

describe("HideOnScroll", (): void => {
	it("mock test", () => {
		render(
			<HideOnScroll>
				<div>basic HideOnScroll</div>
			</HideOnScroll>
		);
		screen.debug();
	});
	// it("should render basic HideOnScroll", () => {
	// 	jest.spyOn(mui, 'useScrollTrigger').mockReturnValue(true);
	// 	const { container } = render(
	// 		<HideOnScroll>
	// 			<div>basic HideOnScroll</div>
	// 		</HideOnScroll>
	// 	);

	// 	screen.debug();
	// 	// expect(container).toMatchSnapshot();
	// });

	// it("should render onScroll HideOnScroll", () => {
	// 	useScrollTriggerMock.mockReturnValue(true);
	// 	const { container } = render(
	// 		<HideOnScroll>
	// 			<div>onScroll HideOnScroll</div>
	// 		</HideOnScroll>
	// 	);

	// 	expect(container).toMatchSnapshot();
	// });
});
