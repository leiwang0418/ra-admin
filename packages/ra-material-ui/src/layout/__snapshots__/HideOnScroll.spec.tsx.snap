// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HideOnScroll should render not onScroll HideOnScroll 1`] = `
<div>
  <div>
    basic HideOnScroll
  </div>
</div>
`;

exports[`HideOnScroll should render onScroll HideOnScroll 1`] = `
<div>
  <div
    style="visibility: hidden; webkit-transform: translateY(-0px); transform: translateY(-0px);"
  >
    basic HideOnScroll
  </div>
</div>
`;
